ui/packages/consul-ui/app/routing/route.js

```diff
@@ -16,36 +16,41 @@ export default class BaseRoute extends Route {
 
   _setRouteName() {
     super._setRouteName(...arguments);
-    const routeName = this.routeName
-      .split('.')
-      .filter(item => item !== 'index')
-      .join('.');
-    const template = get(routes, `${routeName}._options.template`);
-    if (template) {
+
+    const template = get(routes, `${this.routeName}._options.template`);
+    if (typeof template !== 'undefined') {
       this.templateName = template;
     }
-    const queryParams = get(routes, `${routeName}._options.queryParams`);
-    if (
-      queryParams &&
-      ['dc.partitions.index', 'dc.nspaces.index', 'oauth-provider-debug'].includes(this.routeName)
-    ) {
+
+    const queryParams = get(routes, `${this.routeName}._options.queryParams`);
+    if (typeof queryParams !== 'undefined') {
       this.queryParams = queryParams;
     }
   }
 
   redirect(model, transition) {
-    // remove any references to index as it is the same as the root routeName
-    const routeName = this.routeName
-      .split('.')
-      .filter(item => item !== 'index')
-      .join('.');
-    const to = get(routes, `${routeName}._options.redirect`);
+    let to = get(routes, `${this.routeName}._options.redirect`);
     if (typeof to !== 'undefined') {
+      // simple path resolve
+      to = to
+        .split('/')
+        .reduce((prev, item, i, items) => {
+          if (item !== '.') {
+            if (item === '..') {
+              prev.pop();
+            } else if (item !== '' || i === items.length - 1) {
+              prev.push(item);
+            }
+          }
+          return prev;
+        }, this.routeName.split('.'))
+        .join('.');
+      console.log(to);
```

Comment:
remove console.log or is it intentional? 

could add https://eslint.org/docs/rules/no-console separately, just a suggestion

Reply:
:doh: Wow, thanks for spotting! Definitely not intentional, will fix up now.

Weirdly, there should be a eslint rule preventing this, will have a look at that separately.

Reply:
K thanks again @amyrlam this should be good to go now. I did a quick check on why console logs aren't being caught and I've figured why. There'll be a PR coming up with that soonish 🙇‍♂️